compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/tower/TowerResolver.kt

```diff
@@ -227,9 +229,16 @@ class TowerResolver {
             }
 
             // extension for implicit receiver
-            for (nonLocalLevel in nonLocalLevels) {
-                TowerData.BothTowerLevelAndImplicitReceiver(nonLocalLevel, implicitReceiver).process()?.let { return it }
-            }
+            val receiverValue = implicitReceiver.receiverValue
+            val resolveExtensions = receiverValue !is ImplicitClassReceiver ||
```

Comment:
The intentions will look much cleaner if you'll reverse the flag meaning, i.e. name it `skipResolveExtensions` and rewrite the code accordingly.